src/safeds_runner/server/pipeline_manager.py

```diff
@@ -0,0 +1,289 @@
+
+        Parameters
+        ----------
+        websocket_connection : simple_websocket.Server
+            New websocket connection.
+        """
+        self.websocket_target = websocket_connection
+
+    def execute_pipeline(
+        self,
+        pipeline: MessageDataProgram,
+        execution_id: str,
+    ) -> None:
+        """
+        Run a Safe-DS pipeline.
+
+        Parameters
+        ----------
+        pipeline : MessageDataProgram
+            Message object that contains the information to run a pipeline.
+        execution_id : str
+            Unique ID to identify this execution.
+        """
+        if execution_id not in self.placeholder_map:
+            self.placeholder_map[execution_id] = self.multiprocessing_manager.dict()
+        process = PipelineProcess(
+            pipeline,
+            execution_id,
+            self.messages_queue,
+            self.placeholder_map[execution_id],
+        )
+        process.execute()
+
+    def get_placeholder(self, execution_id: str, placeholder_name: str) -> tuple[str | None, Any]:
+        """
+        Get a placeholder type and value for an execution id and placeholder name.
+
+        Parameters
+        ----------
+        execution_id : str
+            Unique ID identifying the execution in which the placeholder was calculated.
+        placeholder_name : str
+            Name of the placeholder.
+
+        Returns
+        -------
+        tuple[str | None, Any]
+            Tuple containing placeholder type and placeholder value, or (None, None) if the placeholder does not exist.
+        """
+        if execution_id not in self.placeholder_map:
+            return None, None
+        if placeholder_name not in self.placeholder_map[execution_id]:
+            return None, None
+        value = self.placeholder_map[execution_id][placeholder_name]
+        return _get_placeholder_type(value), value
+
+
+class PipelineProcess:
+    """A process that executes a Safe-DS pipeline."""
+
+    def __init__(
+        self,
+        pipeline: MessageDataProgram,
+        execution_id: str,
+        messages_queue: queue.Queue[Message],
+        placeholder_map: dict[str, Any],
+    ):
+        """
+        Create a new process which will execute the given pipeline, when started.
+
+        Parameters
+        ----------
+        pipeline : MessageDataProgram
+            Message object that contains the information to run a pipeline.
+        execution_id : str
+            Unique ID to identify this process.
+        messages_queue : queue.Queue[Message]
+            A queue to write outgoing messages to.
+        placeholder_map : dict[str, Any]
+            A map to save calculated placeholders in.
+        """
+        self.pipeline = pipeline
+        self.id = execution_id
+        self.messages_queue = messages_queue
+        self.placeholder_map = placeholder_map
+        self.process = multiprocessing.Process(target=self._execute, daemon=True)
```

Comment:
Add leading `_` to the names of attributes that should be internal to the class.